q2_demux/test/test_demux.py

```diff
@@ -401,3 +404,48 @@ def test_barcode_trimming(self):
         act_metadata = list(actual.metadata.view(YamlFormat).open())
         exp_metadata = ["{phred-offset: 33}\n"]
         self.assertEqual(act_metadata, exp_metadata)
+
+
+class SummaryTests(unittest.TestCase):
+
+    def setUp(self):
+        barcodes = [
+            skbio.DNA('AAAA', metadata={'id': 's1/2', 'description': 'abc/2'},
+                      positional_metadata={'quality': [22, 25, 22, 18]}),
+            skbio.DNA('AAAA', metadata={'id': 's2/2', 'description': 'abc/2'},
+                      positional_metadata={'quality': [25, 25, 25, 25]}),
+            skbio.DNA('AACC', metadata={'id': 's3/2', 'description': 'abc/2'},
+                      positional_metadata={'quality': [22, 25, 22, 18]}),
+            skbio.DNA('AACC', metadata={'id': 's4/2', 'description': 'abc/2'},
+                      positional_metadata={'quality': [22, 25, 22, 18]}),
+        ]
+        sequences = [
+            skbio.DNA('GGG', metadata={'id': 's1/1', 'description': 'abc/1'},
+                      positional_metadata={'quality':
+                      np.array([22, 25, 22])}),
+            skbio.DNA('CCC', metadata={'id': 's2/1', 'description': 'abc/1'},
+                      positional_metadata={'quality':
+                      np.array([29, 29, 29])}),
+            skbio.DNA('AAA', metadata={'id': 's3/1', 'description': 'abc/1'},
+                      positional_metadata={'quality':
+                      np.array([22, 20, 22])}),
+            skbio.DNA('TTT', metadata={'id': 's4/1', 'description': 'abc/1'},
+                      positional_metadata={'quality':
+                      np.array([22, 25, 22])}),
+        ]
+        bsi = BarcodeSequenceIterator(barcodes, sequences)
+
+        barcode_map = pd.Series(['AAAA', 'AACC'], index=['sample1', 'sample2'])
+        barcode_map = qiime.MetadataCategory(barcode_map)
+
+        self.demux_data = emp(bsi, barcode_map)
+
+    def test_basic(self):
+        # test that an index.html file is created and that it has size > 0
+        with tempfile.TemporaryDirectory() as output_dir:
+            result = summary(output_dir, self.demux_data)
+
+            self.assertTrue(result is None)
+            index_fp = os.path.join(output_dir, 'index.html')
+            self.assertTrue(os.path.exists(index_fp))
+            self.assertTrue(os.stat(index_fp).st_size > 0)
```

Comment:
Can you also assert that the png, pdf, and csv files are present in the output dir?